open-craft/sprints
# [BB-1547] Add single-user view

This introduces the whole new view for the user - the single-user view, which shows the assignments for the particular user. To make the user-experience smooth here, some bigger reworks were required, i.e.:
- moving the cell/board data to Redux and caching it in Local Storage - thanks to this we can now browse smoothly between the dashboards and users, while the data is being fetched in the background and the dashboard is being updated dynamically,
- adding issues to the dashboard's serializer, because we originally had them on the backend side and we don't need to fetch them for each user (we're now just doing it to maintain the up-to-date status of the board).

## Testing instructions:
1. Set `.env` file.
1. Set `JIRA_URL` and `REACT_APP_GOOGLE_CLIENT_ID` in `frontend/.env.local` file.
1. Start the backend with `docker-compose -f local.yml up --build`.
1. Navigate to `frontend` and run `npm i && npm start`.
1. Log into the frontend.
1. Open the cell's board and wait for it to fetch.
1. Go to your user's board and count that the values match the ones from the cell's board (and the actual state of your commitments).
1. Go back and forth between different views to confirm that the cache works properly.
